### PR TITLE
Update dev instructions

### PR DESCRIPTION
**Pull request by bonniegee**
## Description

This makes several changes to the README, including:

* Adding a Requirements section, which contains information on poetry and pyenv
* Adding a Quick Start section with instructions on how to get the app up and running for development
* Separating out instructions for locally running Elasticsearch and Postgres
* Adding a Running in Production section, which contains the Installation section
* Adding a development section, which contains sections on testing, code style, CI, and the Git branch workflow

I was unable to run the Elasticsearch Docker instructions on my Mac (`docker: no matching manifest for linux/arm64/v8 in the manifest list entries`), so I copied `Dockerfile-es654.arm64` and `docker-entrypoint-es654-arm.sh` from the [NYPL circulation repository](https://github.com/NYPL-Simplified/circulation/tree/develop/docker). I also made a note about using these on M1 in the Quick Start.

I made some additional changes (primarily changing `-` to `*`) to original text in the README that was flagged by the precommit Markdown linter.

## Motivation and Context

The changes to the README (and additional Docker file) were made while setting up the install locally. Adding a Quick Start section consolidates information related to get an install up and running quickly. Additional sections were added in order to group together information to make it easier for someone who is unfamiliar with the project to find what they're looking for.

## How Has This Been Tested?

Most of these changes are to the README, and I was able to get the app running locally using the Docker file i've added. Additionally, the additional Docker file is _only_ used when running the app on a Mac using M1.

## Checklist

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: docker-entrypoint-es654-arm.sh [hunk 1]
```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
```
Comment (dbernstein) [inline docker-entrypoint-es654-arm.sh:1]: Same as above:  try running circ manager with standard Elastic 7.x docker container.

**Review by dbernstein — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -9,200 +9,154 @@
...
+This will create a new local virtual environment called `circ` that uses Python 3.9.9. Switch to that environment:
 
 ```sh
-pacman -S python
+pyenv local circ
 ```
 
-#### pyenv
-
-[pyenv](https://github.com/pyenv/pyenv) pyenv lets you easily switch between multiple versions of Python. It can be
```
[the remark this answers is not in the document]
Comment (bonniegee) [reply README.md:69]: This was an oversight! I'll add it back.

**Comment by bonniegee** [comment]
@dbernstein I am unable to get the circ manager running with Elasticsearch 7.x. I ran those three commands (run/create the container, add the plugin, restart) and started up the app. This is the error I'm getting in the browser:
```
Could not fetch data: http://localhost:6500/test/groups?max_cache_age=0
{"type": "http://librarysimplified.org/terms/problem/remote-integration-failed", "title": "Third-party service failed.", "status": 502, "detail": "The search index for this site is not properly configured."}
```
When I go to http://localhost:9200/ when the Elasticsearch 7.x container is running, I see the following:
```
{
  "name" : "d2742593c291",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "ksGRilKwRTmmJbNN4BwK9Q",
  "version" : {
    "number" : "7.17.5",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "8d61b4f7ddf931f219e3745f295ed2bbc50c8e84",
    "build_date" : "2022-06-23T21:57:28.736740635Z",
    "build_snapshot" : false,
    "lucene_version" : "8.11.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```

I do not get this error when using 6.x/the Dockerfile. I'm not sure 100% if it's a compatibility issue or if there was a step that I missed.

**Comment by dbernstein** [comment]
Hi @bonniegee :  I looked into it myself.  And yes there does appear to be a compatibility issue with 7.x.   For the time being then, your solution of supplying a docker file for ARM seems like the best available solution until we upgrade to 7.x.  Please make a note in the arm docker file that it can be removed once Elastic is upgraded to 7.x since there are docker arm builds.

**Comment by dbernstein** [comment]
Actually scratch that:  there does appear to be a flag that can be added to the elastic calls that will enable backwards compatibility.  Testing it now.

**Comment by dbernstein** [comment]
@bonniegee : Try running the ES image and circulation code  specified here: https://github.com/ThePalaceProject/circulation/pull/496 .  If it works that please add a note for OSX M1 users.

**Comment by dbernstein** [comment]
@bonniegee :  just a note:  I pushed up another change which I tested but hadn't pushed when I posted the previous note.  The referenced PR (ie https://github.com/ThePalaceProject/circulation/pull/496)  is complete now.

**Comment by bonniegee** [comment]
@dbernstein I pulled down the branch at https://github.com/dbernstein/circulation/tree/make-compatible-with-elasticsearch-7.x, and I'm running into a long error in the browser, screenshots attached (happy to put this comment in the other PR): 

Screen Shot 2022-08-22 at 6 03 01 PM
Screen Shot 2022-08-22 at 6 05 46 PM
Screen Shot 2022-08-22 at 6 06 03 PM

**Comment by jonathangreen** [comment]
HI @bonniegee. I'm sorry this PR has sat open for so long. I believe your issues running the Palace CM code on a M1 mac should be resolved by https://github.com/ThePalaceProject/circulation/pull/411 which lets us support Opensearch 1.x, which builds a container for ARM architecture. 

It seems like this PR has gotten a bit stale, so I'm going to close it, but if you have any problem running on a M1 mac now, feel free to open another issue.

**Pull request closed without merging**